Cargo.toml

```diff
@@ -37,7 +37,7 @@ global-context-less-secure = ["global-context"]
 secp256k1-sys = { version = "0.10.0", default-features = false, path = "./secp256k1-sys" }
 serde = { version = "1.0.103", default-features = false, optional = true }
 
-hashes = { package = "bitcoin_hashes", version = ">= 0.12, <= 0.14", default-features = false, optional = true }
+hashes = { package = "bitcoin_hashes", version = "0.14", default-features = false, optional = true }
```

Comment:
I believe moving the lower bound from 12 to 13 would've been enough - 0.13 already uses `hex-conservative`. But maybe it requires certain version to work so in that case we would need:

```toml
[features]
hashes = ["dep:hashes", "dep:hex-conservative"]

[dependencies]
hex-conservative = { version = ">=0.?", optional = true }
hashes = { package = "bitcoin_hashes", version = ">= 0.13, <= 0.14", default-features = false, optional = true }
```

Where `?` in the version would be replaced with the one that had truncating implemented already. Note however that ranged version was broken by #723 so we have to fix it too if we want ranged versions.

But also, we have hex encoding/decoding in the crate anyway, so why not just have it directly as a dependency and use it for all formatting/decoding? Maybe optional with it being off making `Debug` slower and `Display`/`FromStr` not present.

Reply:
To be clear, this is not why I've requested changes, you don't have to do it. It's just the wrong formatting that's a problem.